src/goto-programs/lazy_goto_model.cpp

```diff
@@ -135,8 +134,10 @@ void lazy_goto_modelt::initialize(const cmdlinet &cmdline)
         source_locationt location;
         location.set_file(filename);
         msg.error().source_location=location;
-        msg.error() << "failed to figure out type of file" << messaget::eom;
-        throw 0;
+        throw deserialization_exceptiont(
+          "failed to figure out type of file"
+          "\nsource location: " +
```

Comment:
A structured variant would be good to enable proper JSON/XML output of the source location.

Reply:
In this case the source location is actually just used to communicate the file name, so I think I'll just reword the message